### ISSUE TITLE
Provide a localized message for screenreaders for TextFields when character limit is exceeded

### ISSUE THREAD

**Issue by rami-a** [flutter/flutter#54279]
Internal: b/150477021

## Use case

When the number of characters exceeds the limit for say a TextFormField, it would be useful to have a standard localized message to communicate to how many characters the user is over by. Otherwise, for an A11y user it can be very difficult to understand what they have to do to correct the error unless we manually code this into our input fields.

## Proposal

Add a localized message that the screenreader announces when the character limit is exceeded in a textfield

cc @goderbauer @xster

**Comment by goderbauer** [comment]
Is this a request to have this on iOS? On Android we are already using the appropriate a11y APIs to communicate the limit.

Also: What's the expected behavior? What kind of announcement are you expecting?

**Comment by rami-a** [comment]
Yes it would be to have it on iOS. Would it be possible to make the behavior match Android's?
